prov/rxd/src/rxd_init.c

```diff
@@ -48,7 +46,19 @@ struct rxd_env rxd_env = {
 };
 
 char *rxd_pkt_type_str[] = {
-	RXD_PKT_TYPES(RXD_OFI_STR_EX, OFI_STR)
+	[RXD_MSG]		= "RXD_MSG",
+	[RXD_TAGGED]		= "RXD_TAGGED",
+	[RXD_READ_REQ]		= "RXD_READ_REQ",
+	[RXD_WRITE]		= "RXD_WRITE",
+	[RXD_ATOMIC]		= "RXD_ATOMIC",
+	[RXD_ATOMIC_FETCH]	= "RXD_ATOMIC_FETCH",
+	[RXD_ATOMIC_COMPARE]	= "RXD_ATOMIC_COMPARE",
+	[RXD_RTS]		= "RXD_RTS",
+	[RXD_CTS]		= "RXD_CTS",
+	[RXD_ACK]		= "RXD_ACK",
+	[RXD_DATA]		= "RXD_DATA",
+	[RXD_DATA_READ]		= "RXD_DATA_READ",
+	[RXD_NO_OP]		= "RXD_NO_OP",
```

Comment:
You could also look at hook_perf.h and hook_perf.c.  That uses macros to do what I think you want to do here -- define an enum and a string array for that enum.  Look for HOOK_FOREACH.

Reply:
That's the approach the code already takes but it wasn't working because it's missing two enums and get's off on the foreach while iterating through the rxd types.